onboarding/source/amb_light_service.c

```diff
@@ -18,22 +25,57 @@ static void lightServiceTask(void * pvParameters);
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned = pdFAIL;
+
+    // Create light task
+    if (lightServiceHandle == NULL) {
+        xReturned = xTaskCreate(lightServiceTask,
+                                LIGHT_SERVICE_NAME,
+                                LIGHT_SERVICE_STACK_SIZE,
+                                NULL,
+                                LIGHT_SERVICE_PRIORITY,
+                                &lightServiceHandle);
+    }
+    
+    if (xReturned == pdFAIL) {
+            printf("\nlightServiceTask not created...\n");
+    }
+
+    // Create queue
+    if (lightServiceQueue == NULL) {
+        lightServiceQueue = xQueueCreate(LIGHT_QUEUE_LENGTH, LIGHT_QUEUE_LENGTH);
+    }
 
     /* USER CODE END */
     return 1;
 }
 
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
+    adcData_t adc_data;
+    adcData_t *adc_data_ptr = &adc_data;
+
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    while (1) {
+        if (xQueueReceive(lightServiceQueue, NULL, (TickType_t) 11) == pdPASS) {
+            adcStartConversion(adcREG1, adcGROUP1);
+            while (!adcIsConversionComplete(adcREG1, adcGROUP1));
+            adcGetData(adcREG1, adcGROUP1, adc_data_ptr);
+
+            unsigned char txt[2];
+            txt[0] = adc_data.value && 0xFF;     
```

Comment:
adc_data_ptr->value returns the ADC value we're interested in. It's a uint16_t. It looks like you're separating the upper and lower bytes of the value, implicitly casting them to chars when you store them in `txt`, and printing `txt`. For example, imagine adc_data_ptr->value is 10. Then, txt[0] = `0b1010` = '\n' and text[1] = `0` = '\0' as those numbers correspond to these characters in the ASCII table. You'll be printing `\n\0` instead of `10`.

Reply:
Look into using `snprintf` instead